text/0268-acceptance-testing-refactor.md

```diff
@@ -0,0 +1,278 @@
+    of tests
+* add routing related helpers
+  * setup importable `visit` method to visit the given url
+  * setup importable `currentRouteName` method which returns the current route name
+  * setup importable `currentURL` method which returns the current URL
+* add DOM interaction helpers (heavily influenced by @cibernox's lovely addon [ember-native-dom-helpers](https://github.com/cibernox/ember-native-dom-helpers))
+  * setup a getter for `this.element` which returns the DOM element representing
+    the applications root element
+  * setup importable `click` helper method
+  * setup importable `tap` helper method
+  * setup importable `triggerKeyEvent` helper method
+  * setup importable `triggerEvent` helper method
+  * setup importable `fillIn` helper method
+  * setup importable `focus` helper method
+  * setup importable `blur` helper method
+  * setup importable `waitUntil` helper method
+  * setup importable `waitFor` helper method
+
+### `setupRenderingTest`
+
+The `setupRenderingTest` function proposed in
+[emberjs/rfcs#232](https://github.com/emberjs/rfcs/blob/master/text/0232-simplify-qunit-testing-api.md)
+(and implemented in
+[ember-qunit](https://github.com/emberjs/ember-qunit)@3.0.0) will be modified to add the same DOM interaction helpers mentioned above:
+
+* setup importable `click` helper method
+* setup importable `tap` helper method
+* setup importable `triggerKeyEvent` helper method
+* setup importable `triggerEvent` helper method
+* setup importable `fillIn` helper method
+* setup importable `focus` helper method
+* setup importable `blur` helper method
+* setup importable `waitUntil` helper method
+* setup importable `waitFor` helper method
+
+Once implemented, `setupRenderingTest` and `setupAcceptanceTest` will diverge from each other in very few ways.
+
+## Changes from Current System
+
+Here is a brief list of the more important but possibly understated changes
+being proposed here:
+
+* The global test helpers that exist now, will no longer be present (e.g.
+  `click`, `visit`, etc) and instead will be available on the test context as
+  well as importable helpers.
+* `this.owner` will now be present and allow (for the first time 🎉) overriding
+  items in the container/registry.
+* The new system will leverage the `Ember.Application` /
+  `Ember.ApplicationInstance` split so that we can avoid creating an
+  `Ember.Application` instance per-test, and instead leverage the same system
+  that FastBoot itself uses to avoid running initializers for each acceptance
+  test.
+* Implicit promise chaining will no longer be present. If your test needs to
+  wait for a given promise, it should use `await`.
+* The test helpers that are included by a new default ember-cli app will be no
+  longer needed and will be removed from the new application blueprint. This
+  includes:
+  * `tests/helpers/resolver.js`
+  * `tests/helpers/start-app.js`
+  * `tests/helpers/destroy-app.js`
+  * `tests/helpers/module-for-acceptance.js`
```

Comment:
Does tests/test-helper.js remain?

Reply:
@kategengler - Yes, though IMHO, it would be better to rename it to `tests/index.js`.  It truly is the entry point to all the tests (and even more obvious now that you have to manually call `start()` to kick off the tests)...

Reply:
Many folks use `module-for-acceptance` to start/stop Mirage servers. So the new blueprint wont include that module helper, but nothing is stopping users from keeping or adding it back, correct?

Reply:
with the new system you would call something like `setupMirage(hooks)` and it would handle starting and stopping of the server in the right hooks automatically for you